t/Util.pm

```diff
@@ -603,33 +603,44 @@ sub read_with_timeout {
 }
 
 sub spawn_forked {
-    my ($code) = @_;
+    my ($code, $opts) = @_;
+    $opts = +{
+        stdout => 1,
+        stderr => 0,
+        %{ $opts || +{} }
+    };
 
-    my ($cout, $pin);
-    pipe($pin, $cout);
-    my ($cerr, $pin2);
-    pipe($pin2, $cerr);
+    my ($outfh, $errfh);
+    if ($opts->{stdout}) {
+        ($outfh, undef) = tempfile(UNLINK => 1);
+    }
+    if ($opts->{stderr}) {
+        ($errfh, undef) = tempfile(UNLINK => 1);
+    }
 
     my $pid = fork;
     if ($pid) {
-        close $cout;
-        close $cerr;
         my $guard = make_guard(sub {
             return unless defined $pid;
             kill 'TERM', $pid;
             while (waitpid($pid, 0) != $pid) {}
         });
         return +{
             pid => $pid,
-            kill => sub { undef $guard; },
-            stdout => $pin,
-            stderr => $pin2,
+            kill => sub {
+                seek $outfh, 0, 0 if $outfh;
+                seek $errfh, 0, 0 if $errfh;
+                undef $guard;
+                ($outfh, $errfh)
```

Comment:
Wouldn't it be simpler if we returned the contents of stdout / stderr rather than their file handles?

Assuming that makes sense, I might recommend changing the code to:
* call `tempdir(CLEANUP => 1)`,
* redirect STDOUT to tempdir/out and optionally STDERR to tempdir/err, and
* in the `kill` subref, read these two files.

Reply:
Addressed in https://github.com/h2o/h2o/pull/3273/commits/d5261ca5b1b5f615f73ec04bb49dc470b797fafa and https://github.com/h2o/h2o/pull/3273/commits/52164f7457124908ac517f8a629ec43bd2c64d64.